docs/design/rubygems.md

```diff
@@ -0,0 +1,344 @@
+
+```Gemfile.lock
+...
+PATH
+  remote: ../foobar
+  specs:
+    foobar (0.1.0)
+...
+```
+
+All path dependencies must be in the project directory.
+Bundler [does not copy](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/source/path.rb#L83)
+those dependencies that are already within the root directory of the project.
+
+#### Plugins
+
+Not supported by cachi2.
+
+### Checksums
+
+Checksum validation is enabled by default.
+It can be disabled with the `BUNDLE_DISABLE_CHECKSUM_VALIDATION` environment variable.
+
+There is also an option to generate checksums in `Gemfile.lock`, but in very weird way.
+(At least, I have not found any other way to do it.)
+
+```shell
+# manually add `CHECKSUMS` section somewhere to the Gemfile.lock
+vim Gemfile.lock
+# install any gem
+bundle add rails --version "6.1.7"
+# check the Gemfile.lock /o\
+cat Gemfile.lock
+```
+
+Example of a checksum section in the `Gemfile.lock` file from my custom [repository](https://github.com/slimreaper35/cachi2-rubygems.git):
+
+```Gemfile.lock
+...
+DEPENDENCIES
+  rails (= 6.1.7)
+
+CHECKSUMS
+  actioncable (6.1.7) sha256=ee5345e1ac0a9ec24af8d21d46d6e8d85dd76b28b14ab60929c2da3e7d5bfe64
+  actionmailbox (6.1.7) sha256=c4364381e724b39eee3381e6eb3fdc80f121ac9a53dea3fd9ef687a9040b8a08
+  actionmailer (6.1.7) sha256=5561c298a13e6d43eb71098be366f59be51470358e6e6e49ebaaf43502906fa4
+  actionpack (6.1.7) sha256=3a8580e3721757371328906f953b332d5c95bd56a1e4f344b3fee5d55dc1cf37
+  actiontext (6.1.7) sha256=c5d3af4168619923d0ff661207215face3e03f7a04c083b5d347f190f639798e
+  actionview (6.1.7) sha256=c166e890d2933ffbb6eb2a2eac1b54f03890e33b8b7269503af848db88afc8d4
+  ...
+
+BUNDLED WITH
+   2.5.14
+```
+
+I believe this feature is available since bundler [v2.5.0](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/lockfile_parser.rb#L55)
+from this [PR](https://github.com/rubygems/rubygems/pull/6374) being merged on Oct 21, 2023.
+
+### Environment variables
+
+The order of precedence for bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environmental variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+#### Relevant environment variables
+
+```txt
+BUNDLE_FORCE_RUBY_PLATFORM=true
+BUNDLE_CACHE_ALL=true
+BUNDLE_CACHE_PATH=${output_dir}/deps/rubygems
+BUNDLE_DISABLE_CHECKSUM_VALIDATION=false
+```
+
+**BUNDLE_CACHE_ALL**: Cache all gems, including path and git gems.
+This needs to be explicitly configured on bundler 1 and bundler 2, but will be the default on bundler 3.
+
+**BUNDLE_CACHE_PATH**: The directory that bundler will place cached gems in when running bundle package,
+and that bundler will look in when installing gems. Defaults to `vendor/cache`.
+
+**BUNDLE_FORCE_RUBY_PLATFORM**: Ignore the current machine's platform and install only ruby platform gems.
+As a result, gems with native extensions will be compiled from source.
```

Comment:
Speaking of platforms, that would be worth describing in more detail. Cachito only supports the pure Ruby platform, I think it's mainly because gems for other platforms contain precompiled binaries and we didn't want to allow those